src/web/js/google-apis/drive.js

```diff
@@ -349,40 +278,15 @@ function createProgramCollectionAPI(clientId, apiKey, collectionName, immediate)
     return createAPI(baseCollection);
   }
 
-  var reauth = function(immediate) {
-    var d = Q.defer();
-    if(!immediate) {
-      // Need to do a login to get a cookie for this user; do it in a popup
-      var w = window.open("/login?redirect=" + encodeURIComponent("/close.html"));
-      window.addEventListener('message', function(e) {
-        if (e.domain === document.location.origin) {
-          d.resolve(reauth(true));
-        } else {
-          d.resolve(null);
-        }
-      });
-    }
-    else {
-      // The user is logged in, but needs an access token from our server
-      var newToken = $.ajax("/getAccessToken", { method: "get", datatype: "json" });
-      newToken.then(function(t) {
-        gapi.auth.setToken({access_token: t.access_token});
-        d.resolve({access_token: t.access_token});
-      });
-      newToken.fail(function(t) {
-        d.resolve(null);
-      });
-    }
-    return d.promise;
-  };
-
-  var initialAuth = reauth(immediate);
-  return initialAuth.then(function(_) {
-    var d = Q.defer();
-    gapi.client.load('drive', 'v2', function() {
-      d.resolve(initialize())
-    });
-    return d.promise;
-  });
-  return initialAuth;
+  var ret = Q.defer();
+  gwrap.load({name: 'drive',
```

Comment:
So the idea is there will be another `gwrap.load()` call for the sheets API, and any other Google APIs we end up using?

Could we load the file picker UI through this?  That's an open feature request (for loading images into the editor) that I'd like to tackle sooner rather than later.